### DISCUSSION TITLE
Graphql Array Result Format Issue

### DISCUSSION THREAD

**Discussion by Passupphp** [rebing/graphql-laravel#922] [General]
**Actual Result :** 

    query GetLanguages{
      getLanguages{
          id
          name
          value
      }
    }

    **Output :** 
    {
    "data": {
        "getLanguages": [
          {
              "id": "10001",
              "name": "English",
              "value": "en",
          }]
        }
    }

**Expected Result :** 

    query GetLanguages{
    getLanguages{
        result {
            id
          name
          value
     }
    }

**Result** : 

    {
    "errors": [
        {
        "message": "Cannot query field \"result\" on type \"Language\".",
        "extensions": {
            "category": "graphql"
        },
        "locations": [
            {
            "line": 3,
            "column": 5
            }
        ]
        }
    ]
    }

**Here my code :** 

    public function fields(): array
    {
        return [
            'id' => [
                'type' => Type::nonNull(Type::string()),
                ],
            'shortName' => [
                'type' => Type::nonNull(Type::string()),
            ],
            'longName' => [
                'type' => Type::nonNull(Type::string()),
            ],
        ];
    }

**Anyone can help to get the expected result ?**

**Comment by mfn** [comment]
Hello @Passupphp , welcome to github and graphql-laravel 😄

You only did show a partial code `public function fields()…` but more would be necessary to better understand your issue. However

> ```
> query GetLanguages{
> getLanguages{
>     result {
>         id
>       name
>       value
>  }
> }
> ```
> 
>…
> `Cannot query field \"result\" on type \"Language\"."`

did you try just with?
```graphql
query GetLanguages{
  getLanguages{
      id
      name
      value
  }
}
```

If it's not working, can you please share the full class for:
- getLanguages query
- the language type (which contains the field)

Thanks!

**Reply by Passupphp** [reply]
Hi @mfn 

Thanks for your response. But i need answer inside the result array

**Type.php**

```php
namespace App\GraphQL\Types;

use App\Models\Language;
use GraphQL;
use GraphQL\Type\Definition\Type;
use Rebing\GraphQL\Support\Type as GraphQLType;

class LanguageType extends GraphQLType
{
    protected $attributes = [
        'name' => 'Language',
        'description' => 'Language details',
        'model' => Language::class
    ];

    public function fields(): array
    {
        return [
            'id' => [
                'type' => Type::nonNull(Type::string()),
            ],
            'name' => [
                'type' => Type::nonNull(Type::string()),
            ],
            'value' => [
                'type' => Type::nonNull(Type::string()),
            ],
            'status' => [
                'type' => Type::nonNull(Type::string()),
            ],
            'createdAt' => [
                'type' => Type::string(),
            ],
            'updatedAt' => [
                'type' => Type::string(),
            ],
        ];
    }
}
```

**Query.php**
```php
namespace App\graphql\Queries;

use App\Models\Language;
use GraphQL\Type\Definition\Type;
use Rebing\GraphQL\Support\Facades\GraphQL;
use Rebing\GraphQL\Support\Query;

class LanguageQuery extends Query
{

    public function type(): Type
    {
        return Type::listOf(GraphQL::type('Language'));
    }

    public function resolve($root, $args)
    {
        return Language::where('status','Active')->get();
    }
}
```
**Query**
```graphql
  query GetLanguages{
  getLanguages{
      result {
          id
        name
        value
   }
  }
```
We need the result inside the results array

**Reply by mfn** [reply]
> But i need answer inside the result array

Then you need to change your type system, if it is that important:
- the `LanguageQuery:type()` should not return `Type::listOf(GraphQL::type('Language'));` but your intermediary `result` type (which you yet would have to create) which then holds the lost of languages.
- And you've to change your `LanguageQuery::resolve()` so its returned "nested" appropriately.

With just this library and no change to the type system, I don't think you can achieve this (ignoring post-JSON-processing middlewares).
